src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -498,6 +530,64 @@ public <N extends Number> void visitUnparsedNumericLiteral(UnparsedNumericLitera
                 FIELD_VALUE, new AstLiteralValue(toBsonValue(unparsedNumericLiteral.getLiteralValue())));
     }
 
+    @Override
+    public void visitSqlSelectionExpression(SqlSelectionExpression sqlSelectionExpression) {
+        sqlSelectionExpression.getSelection().getExpression().accept(this);
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // ORDER BY clause
+
+    @Override
+    public void visitSortSpecification(SortSpecification sortSpecification) {
+        if (sortSpecification.getNullPrecedence() != null
+                && sortSpecification.getNullPrecedence() != NullPrecedence.NONE) {
+            throw new FeatureNotSupportedException(format(
+                    "%s does not support nulls precedence: NULLS %s",
+                    MONGO_DBMS_NAME, sortSpecification.getNullPrecedence()));
+        }
+        if (sortSpecification.isIgnoreCase()) {
+            throw new FeatureNotSupportedException(
+                    format("%s does not support case-insensitive sort key [%s]", MONGO_DBMS_NAME, sortSpecification));
+        }
+        var sortExpression = sortSpecification.getSortExpression();
+        final List<String> fieldPaths;
+        if (sortExpression instanceof SqlTuple sqlTuple) {
+            fieldPaths = acceptAndYield(sqlTuple, FIELD_PATHS);
+        } else {
+            if (!isFieldPathExpression(sortExpression)) {
+                throw new FeatureNotSupportedException(
+                        format("%s does not support sort key not of field path type", MONGO_DBMS_NAME));
+            }
+            var sortFieldPath = acceptAndYield(sortExpression, FIELD_PATH);
+            fieldPaths = singletonList(sortFieldPath);
+        }
+
+        var astSortFields = new ArrayList<AstSortField>(fieldPaths.size());
+        for (var fieldPath : fieldPaths) {
+            var astSortField = new AstSortField(
+                    fieldPath, sortSpecification.getSortOrder() == SortDirection.ASCENDING ? ASC : DESC);
+            astSortFields.add(astSortField);
+        }
+        astVisitorValueHolder.yield(SORT_FIELDS, astSortFields);
+    }
+
+    @Override
+    public void visitTuple(SqlTuple sqlTuple) {
+        List<String> fieldPaths = new ArrayList<>(sqlTuple.getExpressions().size());
+        for (var expression : sqlTuple.getExpressions()) {
+            if (expression instanceof SqlTuple) {
+                fieldPaths.addAll(acceptAndYield(expression, FIELD_PATHS));
+            } else {
+                if (!isFieldPathExpression(expression)) {
+                    throw new FeatureNotSupportedException();
+                }
+                fieldPaths.add(acceptAndYield(expression, FIELD_PATH));
+            }
+        }
+        astVisitorValueHolder.yield(FIELD_PATHS, fieldPaths);
```

Comment:
compared with Hibernate's default `visitTuple` implementation, our implementation is less elegant for we need to return value and currently we only support field path tuple, so the following common value type tuple usage is not supported for now:
```
insert into books (id, title) values (1, "War and Peace"), (2, "Crime and Punishment"), ...
```

Reply:
I refactored the tuple visiting logic to make it generic. Now it will return collection of `Expression`, so it will be solid to work in any scenario.